global_velocity_controller:
  ros__parameters:
    map_frame: map
    base_frame: base_link
    odom_topic: /odom
    path_topic: /plan
    cmd_vel_topic: /cmd_vel
    target_twist_map_topic: /target_twist_map
    global_costmap_topic: "/global_costmap/costmap"

    # PID gains
    kp_xy: 2.5
    ki_xy: 2.0
    kd_xy: 0.5

    kp_yaw: 0.0
    ki_yaw: 0.0
    kd_yaw: 0.0

    # Limits
    max_vx: 3.0
    max_vy: 3.0
    max_wz: 0.0

    anti_windup_integral_limit_xy: 0.5
    anti_windup_integral_limit_yaw: 0.5

    # Anti-oscillation / deadbands
    velocity_deadband: 0.02
    yaw_deadband: 0.02
    stop_speed_threshold: 0.04
    stop_angular_threshold: 0.03

    # Command rate limiting
    cmd_accel_limit_linear: 2.0
    cmd_accel_limit_angular: 4.0 

    # Escape mode params
    # escape_lethal_threshold: 100
    escape_enter_cost_threshold: 99
    escape_target_cost_threshold: 95        # 一直逃到free
    escape_treat_unknown_as_lethal: false
    escape_speed: 0.8
    escape_goal_tolerance: 0.05
    escape_max_radius_cells: 200